src/UmbracoExamine.PDF/ExaminePdfComponent.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Linq;
+using Examine;
+using Examine.LuceneEngine.Providers;
+using Lucene.Net.Index;
+using Umbraco.Core;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Events;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+using Umbraco.Core.Services.Implement;
+using Umbraco.Examine;
+using Umbraco.Web.Scheduling;
+
+namespace UmbracoExamine.PDF
+{
+    /// <summary>
+    ///     Component to index PDF documents in the media library
+    /// </summary>
+    public class ExaminePdfComponent : IComponent
+    {
+        private readonly IExamineManager _examineManager;
+        private readonly ILogger _logger;
+        private readonly PdfIndexCreator _pdfIndexCreator;
+        private readonly PdfIndexPopulator _pdfIndexPopulator;
+
+        public ExaminePdfComponent(
+            IExamineManager examineManager,
+            PdfIndexCreator pdfIndexCreator,
+            ILogger logger,
+            PdfIndexPopulator pdfIndexPopulator)
+        {
+            _examineManager = examineManager;
+            _pdfIndexCreator = pdfIndexCreator;
+            _logger = logger;
+            _pdfIndexPopulator = pdfIndexPopulator;
+        }
+
+        public void Initialize()
+        {
+            foreach (var index in _pdfIndexCreator.Create())
+            {
+                //TODO: This should not be needed but Umbraco needs to update itself to fix the issue
+                // ensure the index is unlocked 
+                if (index is LuceneIndex luceneIndex)
+                {
+                    var dir = luceneIndex.GetLuceneDirectory();
+                    if (IndexWriter.IsLocked(dir))
+                    {
+                        _logger.Info(typeof(ExaminePdfComponent),
+                            "Forcing index {IndexerName} to be unlocked since it was left in a locked state",
+                            luceneIndex.Name);
+                        IndexWriter.Unlock(dir);
+                    }
+                }
+
+                _examineManager.AddIndex(index);
+            }
+
+            //Tie into various events one media saved/trashed/deleted/restored
+            MediaService.Saved += MediaService_Saved;
```

Comment:
These events won't actually work for an index because these events are only ever raised on the admin server but each index exists on each individual server. So for Examine, the events that need to be tied into are the distributed cache events. I'll update.

Reply:
I have fixed this up